### ISSUE TITLE
Add support for per-unit scales

### ISSUE THREAD

**Issue by jesusbotella** [signal-noise/stylelint-scales#46]
> What is the problem you're trying to solve?

Compound validation of scale values with a particular unit is nice, given that units can have different orders of magnitude.

When defining `font-size` with several units, some unexpected combinations can happen. If our rule is defined like this:
```js
{
  "scale": [1, 1.5, 16],
  "units": ["px", "rem"]
}
```

A developer could use `16rem` within styles without raising any validation errors.

> What solution would you like to see?

Probably a compound rule (https://github.com/signal-noise/stylelint-scales/issues/28) (or the current rule) accepting multiple values for the same scale would be good, and will allow developers to avoid those unexpected errors.

**Comment by jeddy3** [comment]
I have a need for this too. I'd like to use a different _space_ scales for `em` and `rem` units. I'd like to enforce a minor third scale for `em` (to align with the [browser defaults](https://html.spec.whatwg.org/multipage/rendering.html) for relative vertical space around type), and use a [powers-of-two](https://styled-system.com/guides/why-powers-of-two/) scale for `rem` for everything else.

The rule can be changed to accept an object as its primary option, e.g.:

```js
{
  "scales/space": {
    "em": [0.58, 0.67, 0.83, 1, 1.17, 1.5, 1.73],
    "rem": [0.25, 0.5, 1, 2, 4, 8]
}
```
